packages/editor/src/client/blocks/richTextBlock/buttons.tsx

```diff
@@ -0,0 +1,234 @@
+      }}
+    />
+  )
+}
+
+interface SlateBlockButtonWithChildrenProps extends ToolbarButtonProps {
+  readonly format: Format
+}
+
+export function FormatButtonWithChildren({format, children}: SlateBlockButtonWithChildrenProps) {
+  const editor = useSlate()
+
+  return (
+    <ToolbarButton
+      active={isFormatActive(editor, format)}
+      onMouseDown={e => {
+        e.preventDefault()
+        toggleFormat(editor, format)
+      }}>
+      {children}
+    </ToolbarButton>
+  )
+}
+
+export function LinkFormatButton() {
+  const editor = useSlate()
+  const [isLinkDialogOpen, setLinkDialogOpen] = useState(false)
+  const [selection, setSelection] = useState<Range | null>(null)
+
+  const [title, setTitle] = useState('')
+  const [url, setURL] = useState('')
+
+  const validatedURL = validateURL(url)
+  const isDisabled = !validatedURL
+
+  const {t} = useTranslation()
+
+  return (
+    <>
+      <ToolbarIconButton
+        icon="link"
+        active={isFormatActive(editor, InlineFormat.Link)}
+        onMouseDown={e => {
+          e.preventDefault()
+
+          const nodes = Array.from(
+            Editor.nodes(editor, {
+              at: editor.selection ?? undefined,
+              match: node => node.type === InlineFormat.Link
+            })
+          )
+
+          const tuple = nodes[0]
+
+          if (tuple) {
+            const [node] = tuple
+
+            setTitle((node.title as string) ?? '')
+            setURL((node.url as string) ?? '')
+          } else {
+            setTitle('')
+            setURL('')
+          }
+
+          setSelection(editor.selection)
+          setLinkDialogOpen(true)
+        }}
+      />
+      <Modal show={isLinkDialogOpen} size={'sm'} onHide={() => setLinkDialogOpen(false)}>
+        <Modal.Body>
+          <Form fluid={true}>
+            <FormGroup>
+              <ControlLabel>{t('blocks.richText.link')}</ControlLabel>
+              <FormControl
+                errorMessage={url && !validatedURL ? 'Invalid Link' : undefined}
+                value={url}
+                onChange={url => setURL(url)}
+              />
+            </FormGroup>
+            <FormGroup>
+              <ControlLabel>{t('blocks.richText.title')}</ControlLabel>
+              <FormControl value={title} onChange={title => setTitle(title)} />
+            </FormGroup>
+          </Form>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button
+            disabled={isDisabled}
+            onClick={() => {
+              insertLink(editor, selection, validatedURL!, title || undefined)
+              setLinkDialogOpen(false)
+            }}>
+            {t('Apply')}
+          </Button>
+          <Button onClick={() => setLinkDialogOpen(false)}>{t('Cancel')}</Button>
+        </Modal.Footer>
+      </Modal>
+    </>
+  )
+}
+
+function validateURL(url: string): string | null {
+  return url
+
+  // TODO: Implement better URL validation with for support relative links.
```

Comment:
Can you create a ticket for this in jira?

Reply:
yep